### PR TITLE
User story 30

### PR DESCRIPTION
**Pull request by ggerdsen**
Added status column to ItemOrder, default value of nil - run DB:migrate
OrdersController's create and update methods now can add or subtract merchant inventory when creating and cancelling orders.
Test should have all of the functionality we need.

### CODE REVIEW

**Comment by NickEdwin** [comment]
Also nice work on that update method, subtracting from and restoring the inventory would be key on a real shops website!

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -11,6 +11,9 @@ def create
     order = user.orders.create(order_params)
     if order.save
       cart.items.each do |item,quantity|
+        new_quantity = (item.inventory - quantity)
```
Comment (sienna-kopf) [inline app/controllers/orders_controller.rb:14]: Nice use of temporary variables here!

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -25,7 +28,21 @@ def create
       render :new
     end
   end
-
+  
+  def update
+    order = Order.where("id = ?", params[:id])
+    item_order = ItemOrder.where("order_id = ?", params[:id])
+    order.update(status: "cancelled")
+    item_order.update(status: "unfulfilled")
```
Comment (sienna-kopf) [inline app/controllers/orders_controller.rb:36]: Nice job changing the status of the order and updating the inventory here!

File: spec/features/orders/cancel_spec.rb [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      orders = Order.where("id = ?", new_order.id)
+      orders.each do |order|
+        expect(order.status).to eq("cancelled")
+      end
+      
+      click_link "My Orders"
+  
+      expect(page).to have_content("Order Number: #{new_order.id}")
+      expect(page).to have_content("Order Status: cancelled")
+
```
Comment (sienna-kopf) [inline spec/features/orders/cancel_spec.rb:92]: Nice thorough test!

**Pull request merged**